tests/performance/asof.xml

```diff
@@ -0,0 +1,29 @@
+<test>
+
+    <preconditions>
+        <table_exists>test.hits</table_exists>
```

Comment:
Need to add more tests, not only for test.hits but also on random data.

Reply:
I'll think of something that makes sense (random but not too random so it's consistent)

Reply:
Added a test that simulates a bunch of visits and origins:

```
SELECT * FROM
                  (
                      SELECT
                          toStartOfDay(now()) + INTERVAL (seconds_since_start_of_day % 86000) SECOND AS date,
                          fingerprint % 50 as fingerprint,
                          multiIf(browserId % 3 == 0, 'firefox', browserId % 3 == 1, 'edge', 'chrome') as browser
                      FROM generateRandom('seconds_since_start_of_day UInt32, fingerprint UInt8, browserId UInt8') LIMIT 100000
                  ) AS origins
                  ASOF LEFT JOIN
                  (
                      SELECT
                          toStartOfDay(now()) + INTERVAL (seconds_since_start_of_day % 86000) SECOND AS date,
                          fingerprint % 50 as fingerprint,
                          multiIf(language % 2 == 0, 'es', 'en') as lang
                      FROM generateRandom('seconds_since_start_of_day UInt32, fingerprint UInt8, language UInt8') LIMIT 5000000
                  ) AS visits
                  ON (visits.fingerprint = origins.fingerprint AND visits.date >= origins.date)
                  FORMAT Null
```

Pre-patch:
```
Queries executed: 100.

localhost:9000, queries 100, QPS: 3.201, RPS: 19502552.116, MiB/s: 111.594, result RPS: 0.000, result MiB/s: 0.000.

0.000%          0.282 sec.
10.000%         0.299 sec.
20.000%         0.302 sec.
30.000%         0.306 sec.
40.000%         0.310 sec.
50.000%         0.313 sec.
60.000%         0.316 sec.
70.000%         0.319 sec.
80.000%         0.322 sec.
90.000%         0.328 sec.
95.000%         0.330 sec.
99.000%         0.335 sec.
99.900%         0.337 sec.
99.990%         0.337 sec.
```

Post-patch:
```
Queries executed: 100.

localhost:9000, queries 100, QPS: 4.828, RPS: 29411511.668, MiB/s: 168.294, result RPS: 0.000, result MiB/s: 0.000.

0.000%          0.188 sec.
10.000%         0.195 sec.
20.000%         0.200 sec.
30.000%         0.202 sec.
40.000%         0.204 sec.
50.000%         0.207 sec.
60.000%         0.209 sec.
70.000%         0.211 sec.
80.000%         0.216 sec.
90.000%         0.219 sec.
95.000%         0.222 sec.
99.000%         0.230 sec.
99.900%         0.236 sec.
99.990%         0.236 sec.
```

Please let me know if you think it needs more performance tests or better approaches.